lib/shared/types/src/reporter.ts

```diff
@@ -0,0 +1,16 @@
+
+export type FlushResults = {
+    successes: number,
+    failures: number,
+    retries: number,
+}
+
+export type Tags = Record<string, string>
+
+export interface DVCReporter {
```

Comment:
Should we change this Reporter interface to be more generic so we don't need to add a new function / type here for every new metric?

Reply:
I tried to keep the amount of data formatting built into the API to a minimum, but yes most of these should be covered by a `reportMetricValue` method with a key.

I think it makes sense to keep `reportFlushResults` as a bespoke call, because the fields on that are deeply related and the api shouldn't predict how they might be used together.